### PR TITLE
Switch api based on version

### PR DESCRIPTION
**Pull request by makyo**

### CODE REVIEW

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/5281/
Test PASSed.

File: jujugui/static/gui/src/app/store/env/legacy-api.js [hunk 1]
```diff
@@ -150,16 +150,16 @@ YUI.add('juju-env-legacy-api', function(Y) {
...
    */
-  function GoEnvironment(config) {
+  function GoLegacyEnvironment(config) {
     // Invoke Base constructor, passing through arguments.
-    GoEnvironment.superclass.constructor.apply(this, arguments);
+    GoLegacyEnvironment.superclass.constructor.apply(this, arguments);
   }
 
-  GoEnvironment.NAME = 'go-env';
+  GoLegacyEnvironment.NAME = 'go-env';
```
Comment (frankban) [inline jujugui/static/gui/src/app/store/env/legacy-api.js]: s/go-env/go-legacy-env/ ? [shrug]

File: jujugui/static/gui/src/app/store/env/legacy-api.js [hunk 2]
```diff
@@ -3256,7 +3256,7 @@ YUI.add('juju-env-legacy-api', function(Y) {
 
   environments.endpointToName = endpointToName;
```
Comment (frankban) [inline jujugui/static/gui/src/app/store/env/legacy-api.js]: I guess we can remove many of these helper functions already defined in api.js.
The legacy API in this case can safely reuse functions from api.js.

File: jujugui/static/gui/src/test/test_app.js [hunk 3]
```diff
@@ -109,6 +109,7 @@ describe('App', function() {
 
     function constructAppInstance(config, context) {
       config = config || {};
+      config.jujuCoreVersion = config.jujuCoreVersion || '2.0.0';
```
Comment (frankban) [inline jujugui/static/gui/src/test/test_app.js]: Yeah version checks in api.js will soon go away.

**Comment by frankban** [comment]
:+1: with a couple of comments, thank you!

**Comment by frankban** [comment]
I am going to merge this because I need it for my current branch, Will do a follow up with changes.
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/juju-gui-merge

**Pull request merged**